plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -271,9 +271,23 @@ private CompletableFuture<ConnectorSplitBatch> getNextBatchInternal(int maxSize)
                 }
 
                 if (recordScannedFiles) {
-                    // Positional and Equality deletes can only be cleaned up if the whole table has been optimized.
-                    // Equality deletes may apply to many files, and position deletes may be grouped together. This makes it difficult to know if they are obsolete.
-                    List<org.apache.iceberg.DeleteFile> fullyAppliedDeletes = tableHandle.getEnforcedPredicate().isAll() ? wholeFileTask.deletes() : ImmutableList.of();
+                    // Equality deletes can only be cleaned up if the whole table has been optimized.
+                    // Equality and position deletes may apply to many files, however position deletes are always local to a partition
```

Comment:
In which situation does a position delete apply to multiple files?

Reply:
I don't think Trino writer does that, but it is something supported by iceberg spec and library https://github.com/apache/iceberg/blob/70c506ebad2dfc6d61b99c05efd59e884282bfa6/core/src/main/java/org/apache/iceberg/deletes/DeleteGranularity.java#L26